recipes/default.rb

```diff
@@ -84,8 +84,11 @@
 Chef::Log.info "Overriding mysql/tunable/binlog_format to 'MIXED'"
 node.override['mysql']['tunable']['binlog_format'] = 'MIXED'
 
-# Convert the server IP to an integer and use it for the server-id attribute in my.cnf
-server_id = RsMysql::Helper.get_server_ip(node).to_i
+# Drop first 2 hex numbers from MAC address to have a 32-bit integer and use it for the server-id attribute in my.cnf.
+# This is used since MAC addresses within the same network must be different to correctly talk to each other.
+# Some clouds use the same public IP with different ports for multiple instances, so IP is not unique.
+server_id = node['macaddress'].split(/\W/)[2..-1].map {|item| item.hex.to_s(2)}.join.to_i(2)
```

Comment:
Let's use `node['macaddress'].split(/\W/)[2..-1].join.to_i(16)` instead since that will directly convert from the later 32 bits of the MAC address into an integer.